tests/functional/rdir/test_server.py

```diff
@@ -666,3 +685,133 @@ def test_volume_lock(self):
         proc1 = subprocess.Popen(['oio-rdir-server', cfg], stderr=fd)
         self.garbage_procs.append(proc1)
         self.assertTrue(check_process_absent(proc1))
+
+
+class TestRdirServer4(RdirTestCase):
+    def setUp(self):
+        super(TestRdirServer4, self).setUp()
+        self.num, self.db_path, self.host, self.port = self.get_service('rdir')
+        self.port = int(self.port)
+        self.vol = self._volume()
+
+    def tearDown(self):
+        super(TestRdirServer4, self).tearDown()
+
+    def test_meta2_fetch_invalid_parameters(self):
+        # fetch without volume
+        resp = self._post("/v1/rdir/meta2/fetch")
+        self.assertEqual(resp.status, 400)
```

Comment:
I suggest posting a body to be sure that the request fails because of the missing volume.

Reply:
Fixed in b612b14